e2e/system_test.go

```diff
@@ -199,3 +200,182 @@ func TestSystem_StakeAmount(t *testing.T) {
 		})
 	}
 }
+
+func TestSystem_UnstakeAmount(t *testing.T) {
```

Comment:
Agreed 👍 
I extracted out two useful methods:
`getAccountBalance` and `getStakedBalance` to avoid code duplication.
Also, I've extracted out the transaction object construction as it can be parametrized